src/GitHub.Api/SimpleApiClientFactory.cs

```diff
@@ -28,10 +28,6 @@ public SimpleApiClientFactory(IProgram program, Lazy<IEnterpriseProbeTask> enter
 
         public ISimpleApiClient Create(UriString repositoryUrl)
         {
-            var contains = cache.ContainsKey(repositoryUrl);
-            if (contains)
-                return cache[repositoryUrl];
-
```

Comment:
I removed the above block because we were accessing `cache` outside of the lock here. In theory, if `ClearFormCache` is called after we check `contains` (line 32) but before we execute the `return` (line 33), we could get an exception here because the item could be removed from the `cache`.

If we're concerned about performance here, we could consider switching to the `ConcurrentDictionary`, but I'm not sure this is a hot spot. Thoughts @shana?